app/src/main/java/com/ecosystem/kin/app/MainActivity.java

```diff
@@ -73,4 +84,29 @@ private void openKinMarketplace() {
             e.printStackTrace();
         }
     }
+
+    private void createNativeSpendOffer() {
+        String offerJwt = JwtUtil.generateSpendOfferExampleJWT(BuildConfig.SAMPLE_APP_ID);
+        Log.d(TAG, "createNativeSpendOffer: " + offerJwt);
+        try {
+            Kin.purchase(offerJwt, new Callback<String>() {
```

Comment:
does call-back called on UI thread?

Reply:
yes

Reply:
This is an example of possible problem of keeping reference to the callback after the Activity destroyed.
The callback will have a reference within Kin and the callback will hold reference to the Activity. We should consider holding weak reference to the callback. We will also need to support a flow when purchase is requested callback is destroyed and a second callback is provided for a new purchase call with the same offer details

Reply:
will make it a weak reference 👍